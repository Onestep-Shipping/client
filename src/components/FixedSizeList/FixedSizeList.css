.page-indicator-container {
  margin-top: 30px;
  width: 100%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-evenly;
  margin-bottom: 30px;
}

.indicator-text {
  user-select: none;
  cursor: pointer;
  font-size: 18px;
  color: white;
}

.indicator-container {
  padding: 7px 10px;
  background-color: #224DAA;
  transition: background-color 0.3s ease-out;
  border-radius: 2px;
}

.empty-placeholder {
  width: 100%;
  display: flex;
  flex-direction: column;
  padding: 50px 0px;
  align-items: center;
  justify-content: center;
  background-color: rgb(240, 240, 240);
  border-radius: 20px;
  color: rgb(179, 0, 0);
  font-weight: 500;
  font-size: 18px;
}