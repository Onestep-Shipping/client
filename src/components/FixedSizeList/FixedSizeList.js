import React, {useState} from 'react';
import './FixedSizeList.css';
import PropTypes from 'prop-types';

const LIST_SIZE = 5;

const FixedSizeList = (props) => {
  const { headers, data, row } = props;
  const [currentPage, setCurrentPage] = useState(1);

  const max = Math.ceil(data.length / LIST_SIZE);

  const handleNext = () => {
    const newPageVal = currentPage + 1;
    if (newPageVal <= max) {
      setCurrentPage(newPageVal);
    }
  }

  const sliceArray = () => {
    if (data.length >= LIST_SIZE) {
      const endIndex = currentPage === max ? data.length : (currentPage * LIST_SIZE);
      return data.slice((currentPage - 1) * LIST_SIZE, endIndex);
    }
    console.log(data);
    return data;
  }

  const handlePrev = () => {
    const newPageVal = currentPage - 1;
    if (newPageVal > 0) {
      setCurrentPage(newPageVal);
    }
  }

  return (
    <div>
      <div className="schedule-result-header-row">
        {headers.map((header, ind) => 
          <div className={"col" + (ind === 0 ? "-numb" : "")} key={ind}>
            <text className="schedule-result-header-text">{header.toUpperCase()}</text>
          </div>
        )}
      </div>
      {sliceArray().length < 1 ?
        <div className="empty-placeholder">NO RESULTS</div> 
        :
        sliceArray().map((booking, ind) => 
        <div key={ind}>
          {row(booking, ((currentPage - 1) * LIST_SIZE) + ind)}
        </div>
      )}
      {data.length > LIST_SIZE &&
      <div className="page-indicator-container">
        <button className="result-button" onClick={handlePrev}>{'<< Previous'}</button>
        <text>Page {currentPage} / {max === 0 ? 1 : max}</text>
        <button className="result-button" onClick={handleNext}>{'Next >>'}</button>
      </div>}
    </div>
  );
}

export default FixedSizeList;

FixedSizeList.propTypes = {
  headers: PropTypes.array,
  data: PropTypes.array,
  row: PropTypes.element,
};
