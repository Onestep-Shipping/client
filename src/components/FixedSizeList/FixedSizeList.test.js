import FixedSizeList from './FixedSizeList';
import React from 'react';
import { shallow } from 'enzyme';

const row = (shipment, ind) => {
  return (<div key={ind} />);
};

const dataSize = 20;
const listSize = 5;

let wrapper = shallow(
  <FixedSizeList 
    headers={Array(dataSize).fill("Headers")} 
    data={Array(dataSize).fill("Data")} 
    row={row}
  />
);

describe('FixedSizeList', () => {
  it('should render the FixedSizeList Component correctly', () => {   
    expect(wrapper).toMatchSnapshot();
  });
  it('should have the correct max page number', () => {  
    wrapper.find("#prev-button").simulate('click');
    expect(
      wrapper.find("#current-page").text().split(" / ")[1]
    ).toEqual((dataSize / listSize).toString());
  });
  it('should not go to previous page if previous button is clicked', () => { 
    expect(wrapper.find("#current-page").text().split(" / ")[0]).toEqual("Page 1");
     
    wrapper.find("#prev-button").simulate('click');
    expect(wrapper.find("#current-page").text().split(" / ")[0]).toEqual("Page 1");
  });
  it('should go to next page if previous button is clicked', () => {  
    wrapper.find("#next-button").simulate('click');
    expect(wrapper.find("#current-page").text().split(" / ")[0]).toEqual("Page 2");

    wrapper.find("#next-button").simulate('click');
    expect(wrapper.find("#current-page").text().split(" / ")[0]).toEqual("Page 3");

    wrapper.find("#next-button").simulate('click');
    expect(wrapper.find("#current-page").text().split(" / ")[0]).toEqual("Page 4");
  });
  it('should not go to previous page if previous button is clicked', () => { 
    expect(wrapper.find("#current-page").text().split(" / ")[0]).toEqual("Page 4");
     
    wrapper.find("#next-button").simulate('click');
    expect(wrapper.find("#current-page").text().split(" / ")[0]).toEqual("Page 4");
  });
  it('should go to previous page if previous button is clicked', () => {  
    wrapper.find("#prev-button").simulate('click');
    expect(wrapper.find("#current-page").text().split(" / ")[0]).toEqual("Page 3");
  });
});