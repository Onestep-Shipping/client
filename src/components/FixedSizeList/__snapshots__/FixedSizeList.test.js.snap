// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixedSizeList should render the FixedSizeList Component correctly 1`] = `
<div>
  <div
    className="schedule-result-header-row"
  >
    <div
      className="col"
      key="0"
    >
      <span
        className="schedule-result-header-text"
      >
        HEADERS
      </span>
    </div>
    <div
      className="col"
      key="1"
    >
      <span
        className="schedule-result-header-text"
      >
        HEADERS
      </span>
    </div>
    <div
      className="col"
      key="2"
    >
      <span
        className="schedule-result-header-text"
      >
        HEADERS
      </span>
    </div>
    <div
      className="col"
      key="3"
    >
      <span
        className="schedule-result-header-text"
      >
        HEADERS
      </span>
    </div>
    <div
      className="col"
      key="4"
    >
      <span
        className="schedule-result-header-text"
      >
        HEADERS
      </span>
    </div>
    <div
      className="col"
      key="5"
    >
      <span
        className="schedule-result-header-text"
      >
        HEADERS
      </span>
    </div>
    <div
      className="col"
      key="6"
    >
      <span
        className="schedule-result-header-text"
      >
        HEADERS
      </span>
    </div>
    <div
      className="col"
      key="7"
    >
      <span
        className="schedule-result-header-text"
      >
        HEADERS
      </span>
    </div>
    <div
      className="col"
      key="8"
    >
      <span
        className="schedule-result-header-text"
      >
        HEADERS
      </span>
    </div>
    <div
      className="col"
      key="9"
    >
      <span
        className="schedule-result-header-text"
      >
        HEADERS
      </span>
    </div>
    <div
      className="col"
      key="10"
    >
      <span
        className="schedule-result-header-text"
      >
        HEADERS
      </span>
    </div>
    <div
      className="col"
      key="11"
    >
      <span
        className="schedule-result-header-text"
      >
        HEADERS
      </span>
    </div>
    <div
      className="col"
      key="12"
    >
      <span
        className="schedule-result-header-text"
      >
        HEADERS
      </span>
    </div>
    <div
      className="col"
      key="13"
    >
      <span
        className="schedule-result-header-text"
      >
        HEADERS
      </span>
    </div>
    <div
      className="col"
      key="14"
    >
      <span
        className="schedule-result-header-text"
      >
        HEADERS
      </span>
    </div>
    <div
      className="col"
      key="15"
    >
      <span
        className="schedule-result-header-text"
      >
        HEADERS
      </span>
    </div>
    <div
      className="col"
      key="16"
    >
      <span
        className="schedule-result-header-text"
      >
        HEADERS
      </span>
    </div>
    <div
      className="col"
      key="17"
    >
      <span
        className="schedule-result-header-text"
      >
        HEADERS
      </span>
    </div>
    <div
      className="col"
      key="18"
    >
      <span
        className="schedule-result-header-text"
      >
        HEADERS
      </span>
    </div>
    <div
      className="col"
      key="19"
    >
      <span
        className="schedule-result-header-text"
      >
        HEADERS
      </span>
    </div>
  </div>
  <div
    key="0"
  >
    <div
      key="0"
    />
  </div>
  <div
    key="1"
  >
    <div
      key="1"
    />
  </div>
  <div
    key="2"
  >
    <div
      key="2"
    />
  </div>
  <div
    key="3"
  >
    <div
      key="3"
    />
  </div>
  <div
    key="4"
  >
    <div
      key="4"
    />
  </div>
  <div
    className="page-indicator-container"
  >
    <button
      className="result-button"
      id="prev-button"
      onClick={[Function]}
    >
      &lt;&lt; Previous
    </button>
    <span
      id="current-page"
    >
      Page 
      1
       / 
      4
    </span>
    <button
      className="result-button"
      id="next-button"
      onClick={[Function]}
    >
      Next &gt;&gt;
    </button>
  </div>
</div>
`;
